### PR TITLE
network: enable GatewayOnLink= if no static address is configured

### PR DESCRIPTION
**Pull request by yuwata**
This also renames `GatewayOnlink=` to `GatewayOnLink=`.

C.f. #1850.

### CODE REVIEW

File: src/network/networkd-network.c [hunk 1]
```diff
@@ -252,19 +252,24 @@ static int network_verify(Network *network) {
...
-                                                 network->filename, network->filename);
-
-        LIST_FOREACH(addresses, address, network->static_addresses)
-                if (!address->family)
-                        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                 "%s: Address section without Address field configured. "
-                                                 "Ignoring %s.",
-                                                 network->filename, network->filename);
+        LIST_FOREACH_SAFE(addresses, address, address_next, network->static_addresses)
+                if (!address->family) {
```
Comment (poettering) [inline src/network/networkd-network.c]: I think `address->family == AF_UNSPEC` would be anicer check here

File: src/network/networkd-network.c [hunk 2]
```diff
@@ -252,19 +252,24 @@ static int network_verify(Network *network) {
...
+                        log_warning("%s: Address section without Address= field configured. "
+                                    "Ignoring [Address] section from line %u.",
+                                    network->filename, address->section->line);
+
+                        address_free(address);
+                }
+
+        LIST_FOREACH_SAFE(routes, route, route_next, network->static_routes)
+                if (!route->family) {
```
Comment (poettering) [inline src/network/networkd-network.c]: dito

**Comment by poettering** [comment]
looks ok to me, two tiny nitpicks, see above. CI doesn't like it though.

**Comment by yuwata** [comment]
Updated.
- `GatewayOnlink=` -> `GatewayOnLink=`.
- enable `GatewayOnLink=` only when it is not specified.
- added several config parse cleanups.

**Comment by yuwata** [comment]
Wow, all green!

**Pull request merged**